### ISSUE TITLE
Type checking `pytest.param`

### ISSUE THREAD

**Issue by jamesbraza** [dbader/pytest-mypy#197]
Imagine I have some code like this:

```python
import pytest


@pytest.mark.parametrize(
    "arg",
    [
        pytest.param("1", id="good"),
        pytest.param(2, id="bad"),
    ],
)
def test_arg(arg: str) -> None:
    assert arg
```

Currently with Python 3.11, `mypy==1.16.0` and `pytest-mypy==1.0.1` won't flag that the second `pytest.param` actually has a type mismatch (`int` vs `str`).

It would be super useful to flag this as a latent `TypeError` when using `pytest --mypy`.

**Comment by dmtucker** [comment]
Hey there 👋 Unfortunately, this isn't something this plugin can deliver. Pytest and/or Mypy need to support it...
For example, if we just run `mypy` on your file, it can't detect this:
```console
$ pipenv install pytest-mypy
$ pipenv run mypy sscce.py 
Success: no issues found in 1 source file
```
Either Pytest needs to ship type-aware parametrizations, or Mypy needs to be enhanced to detect this.
(I would guess the former, perhaps as part of https://github.com/pytest-dev/pytest/issues/3834? I'd be interested to get pytest-dev's thoughts on this.)
